### PR TITLE
Add type definition

### PR DESCRIPTION
**Pull request by 38elements**
This adds type definition.

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  markup: string
+  components?: Components
+  onError?: (error: ErrorParam) => void
+  'allow-scripts'?: boolean
+  'allow-events'?: boolean
+  [key: string]: any
+}
+
+
+class Markup extends Component<Props, {}> {
```
Comment (Zoramite) [inline src/index.d.ts:26]: This can be changed to `export default class ...` instead of adding the export at the end.

Comment (donovantc) [reply src/index.d.ts:26]:
```suggestion
export default class Markup extends Component<Props, {}> {
```

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  static setReviver(h: typeof createElement): void
+  setComponents(components: Components): void
+  render(props: Props, {}): VNode
+}
+
+
+export default Markup
```
Comment (donovantc) [inline src/index.d.ts:33]:
```suggestion
```

**Review by donovantc — COMMENTED** [review]
By making these suggested changes this works, at least based on testing it locally. Additionally, it solves #43.

**Comment by donovantc** [comment]
Any chance you could finalise this PR @38elements? Unfortunately, I'm dependent on these changes and getting this merged would save me having to use a fork of the repo. Thanks.

**Comment by 38elements** [comment]
It has already been completed.
It seems your action is just cosmetic change.

**Comment by donovantc** [comment]
Yeah and based on a suggestion from the reviewer @Zoramite. @developit would you be willing to merge this PR?

**Comment by developit** [comment]
Sorry for the delay. 

FWIW typescript definitions really shouldn't prevent usage of a module. You can declare them in your own project using ambient module definitions:

```ts
module 'preact-markup' {
  export default Markup {
    // Etc
  }
}
```

**Pull request merged**